spectral_fitting/parallel_spectral_fitting.ipynb

```diff
@@ -0,0 +1,1170 @@
+3mZthGCF12i1NzDvOTl2+N6m7AbghQx+fJQ6YC/DBfXm+CbcO599LG4Lz3887R/TuKFxbWZZ3eQ6Knvq6YypoRtFL2jtNqJYWc07V7YqcKuqrwwCbx935M/lng7qiZwpS9s1CyiMYi0+nOM6bWmoYRkAd1dIFIyKDtVvNj83cDMMIaBHf0r8XkT7ihFTfVtV9diQ24WYYRkAjZma1UNXzE/u3k4D/IiLHEfuWfldVb8/Shwm3DqdUngzOx7rcjvzySpiPtDvvVil884/dxdBkZG/k+uwi/Jbv8tbxiqn1OB8/qGU5Vwzq9hSdKUi54tbcoij8XYxFon7uV/uFqlaA+5MDEVkJ/GHW+024GYYR0KobCqq6HfjnrO1NuBmGEdAKamk9MOHW4UwVfxec93a/sFpe0X1uUFfw3mk//2j6VV+W73f9pzwUejw1tTfvl8NeBryHnVB5SVD3XVz0k0plulqOovm8dIx6UcnkftWY5KaJF8MJQNVOSVXvy3KvCTfDMEJaSy29E9gNzGRLigATboZh7DstppaWVfVNC7kxk3ATkdcAFwHvIF7QOxh4QFUvXchDjebhq6knD/xJqra3WlrV717woVQKwMmy01mLUbizlvOU2EMGXHkglYdhr7dBmsuFfZxUek21vL1vT/pXMBaZFttQ2CAi7yN2tgeyq6U193wTW5N1xEsvbwEeUdXTgaUicuKChmsYRssSRfmaRwM5DXgz8J7keHfWG7PM3C4EvgW8HDgZuC25flfy4B/vw0ANw2hxWmzm1q+q5yzkxnmFW5Lg4Txib/yX40XQJI7PNDTLPeuAyxcyGMMwmk+lRYx4E34jIh8AHiHeTKjbbullwCdnOsWLoJn8fM6CSBKJc51/bZaInUYLUEwFQF3d717qw4fcoth4OXzZt0z4pgLht/zRw86joK/gzEQKufAVGCu5V++342Egy8myO++eXuaelAvX/qIo9Gww6kOLzdx6iCdWL0/O67Zbempy9BFHw7wWOJPYHeIs4pDBhmF0EA1eU6vFx1VVZ05E5JKsN84r3FT1zKTDNcCniCNm3iQi9wMPq+oPFzJawzBalxabud0gIv+V2Gr4i8DjWW/MZAqiqhuBP05O37avozNakw8eHn78fYXd1fLqJW7FIZ9SKac805B8PrRU3zvpvBcKXl0lZTs17FnBT5bTMwU3rnLkyq8d+NPn/A5G/WkxO7e3AbcQq6cfUNX7s95oRryGYQRUnvNl03hE5HLcWv/jwDnAOSJyjqp+NEsfJtwMwwio18xNRIaJZ10DwDbgnapaTOo+Q5yMeQq4U1WvSN2+0Ss/BXxvX59vws0wjIA6bihcBNymqteLyEeBN+HsZI8EzpgRdrMwrao3z9WxiLxLVecNf2TCrUM4ZOjV1fIAw9XyXrYF7T655qhq+azjwnSR/Qe4/KE534yjJ3z/Kp4Zx67NBwV127avqJafHXXjKKVsp/zZwUBXGFlk0FOLlvW4tblCzl7XRpBeH90ProNqItsuYNqrWwPcISJdwJ+p6q9S9+4UkW8DPyHOVr8NWAr8AXGawE/Xeri9LYZhBNRrt1RV9wKIyEnAWuBvkvMe4EZiC4xjgWuAc1P3bhCR7xJH3j0DWEEs4O4CPpxE6Z0XE26GYQRkXXObxTh/fWLE77c5FbgaeKOqzliNl4BrVHUaeEhEDpytf1WNgO8kxz5jwq1NWTH4iuB8IO8s+fsrA67MoUG715/idtKXnfKboK4yMvtaS647/JKMPA+F9Le8fz5VciYjO8bDTG0Vz7NhsCv0lBj1si+t7HVjip4TNtNYDLK6X6nqvB+IiBxJLNjOU9WtXtVq4FYROQ04CtiywKHOiwk3wzACKvXbUPgw8TrZLSICcA+wQ1WvFZGvE3s6TQDvq9cDfUy4GYYRUMc1t/fMU3clcGWtPkTkYOI4ktWpv9m5dSBDfVIt9xbCgCw5LzRfwftY37VqSdCuZ3CiWo6OODyoy4+73VKmp5mL3HbnydDt9QfQN+Cl/ZvHcb5QcKpoTyHcLd093ePVOZW4O98SyYI7nhbzUPgq8G/Etm77hAk3wzACWky4TanqJxZyowk3wzAC6mjntmBE5IykuFtELiW2d6uAZb8yDGOBtEiwypn1uhFiW7hjk3PLftWJRLi1qVwq/UXeOz97qTMLee0RYYSY3pUu2kfUFa7HTR9xfLVcGPeigozuDtrli249rmf5SFi3ZaWr89bVlg2MBe1GJvuYi+Fu5xExVnK/V8VCnjaEVpi5zWxGiMh/U9WrZ66LyIez9mHCzTCMgFZYc0tiuF0IHC8if5RczgMHEMeVrIkJN8MwAlpBuBHvkn6PONXB3xLHs48g5Sw9Dybc2oixSedbfODQSUHdH/Y7M5EB33wiZf2PF0Ayv3tnUFU++nXVcqX/GdducG/QzjfJyG95IqjrGXSmIL29Lp9CObWO0+2NMZcyE1nW5/r49agLfvmd8bRvtbEYtIJaSmwADLH3wrtTdZns3Fpi5dAwjNahUsnXPBrAU8lxCrGXw2PEuVyOn+8mH5u5GYYR0AozN1W9Eapx2/4yufwvInJn1j5MuBmGEdAia24zLBGRk4nt3E4D+mu0r2LCrU3ZNvqj4PwtB3ygWl7e617OkbEwGkdp1EUM6ZqaDOro8pK7rHxltVze/mDQLOryEsQMBFV09bs+/fW+8eneoJ2/zpZej9sz5dpum3R1T458E2PxaYWZm8eFwHrgGOAX7EOCKhNuhmEEtEJoKRE5SFWfBUaBS3G7pZkdjE24GYYR0CJq6WXAXxJH7J2Z5s8IuLOydGDCrQOZLLuXc2wq9AQoT7iIG3QHVQwOOXOSiY23u2YjzwbtcpPj7qQ3fIXKk06lzOfndinw/wONelFAAL73rIt4smHysTn7MBaH9DJBM/A2Ef4N+JqqPrmvfZhwMwwjoMXW3LYBV4jIauBe4F9V9adZbmy+iDYMo6WIolzNo1Go6k3AnxInk3kdcTTfTN
```

Comment:
Perhaps my only substantial comment here - why not just use ``apply_function_parallel_spectral`` instead of re-implementing all the looping and chunking logic in this PR?

Reply:
Good point, though I think we'd need to add a "non-cube" return option to `apply_function_parallel_spectral`. Right now it feeds all outputs through `_new_cube_with` and we just want an array returned here.

Reply:
talked about this on the call - I'm in favor of incorporating this into spectral-cube (allowing a non-cube return for some forms of `apply_function`...) and keeping this here as an example of how you would go about developing your own custom-return-object addon to spectral-cube

Reply:
I've cleaned this up to just use `apply_function_parallel_spectral`. This includes a description of handling custom returns, so the crucial part is now just writing the custom function to apply (in this case the spectral fitting function).